staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -255,6 +255,17 @@ func validateCustomResourceDefinitionSpec(spec *apiextensions.CustomResourceDefi
 	allErrs = append(allErrs, validateCustomResourceDefinitionValidation(spec.Validation, mustBeStructural, hasAnyStatusEnabled(spec), opts, fldPath.Child("validation"))...)
 	allErrs = append(allErrs, ValidateCustomResourceDefinitionSubresources(spec.Subresources, fldPath.Child("subresources"))...)
 
+	// If no top-level schema is provided, ensure all individual versions define a schema
+	// v1beta1 defaults missing schemas to a minimal schema for compatibility.
+	// v1+ requires API clients to provide a schema.
+	if spec.Validation == nil {
+		for i, v := range spec.Versions {
+			if v.Schema == nil {
```

Comment:
this is weird because people can create CRDs with empty schema
```
  versions:
  - name: v1
    schema: {}
    served: true
    storage: true
```

Reply:
that's neat. will stew on that.

Reply:
yeah, will require both schema and openapiv3 schema to be set

Reply:
fixed